pkg/stanza/operator/transformer/recombine/recombine.go

```diff
@@ -267,47 +277,71 @@ func (r *Transformer) matchIndicatesLast() bool {
 
 // addToBatch adds the current entry to the current batch of entries that will be combined
 func (r *Transformer) addToBatch(_ context.Context, e *entry.Entry, source string) {
-	if _, ok := r.batchMap[source]; !ok {
-		r.batchMap[source] = []*entry.Entry{e}
+	if batch, ok := r.batchMap[source]; !ok {
+		r.batchMap[source] = &SourceBatch{
+			entries:                 []*entry.Entry{e},
+			recombineFieldTotalSize: 0,
+		}
 		if len(r.batchMap) >= r.maxSources {
 			r.Error("Batched source exceeds max source size. Flushing all batched logs. Consider increasing max_sources parameter")
 			r.flushUncombined(context.Background())
+			return
+		}
+	} else {
+		batch.entries = append(batch.entries, e)
+	}
+	sourceBatch := r.batchMap[source]
+
+	if r.maxLogSize > 0 {
+		var s string
+		err := e.Read(r.combineField, &s)
+		if err != nil {
+			r.Errorf("entry does not contain the combine_field")
+			return
+		}
+		bytesCount := int64(len(s))
+		sourceBatch.recombineFieldTotalSize += bytesCount
```

Comment:
```suggestion
		sourceBatch.recombineFieldTotalSize += int64(len(s))
```
We can avoid this allocation